huaweicloud/services/ccm/resource_huaweicloud_ccm_certificate_push.go

```diff
@@ -0,0 +1,383 @@
+	pushCertificatePath := scmClient.Endpoint + pushCertificateHttpUrl
+	pushCertificatePath = strings.ReplaceAll(pushCertificatePath, "{certificate_id}", d.Get("certificate_id").(string))
+
+	pushCertificateOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"Content-Type": "application/json"},
+	}
+	pushCertificateOpt.JSONBody = utils.RemoveNil(buildPushCertificateBodyParams(d))
+	pushCertificateResp, err := scmClient.Request("POST", pushCertificatePath, &pushCertificateOpt)
+	if err != nil {
+		return diag.Errorf("error push certificate: %s", err)
+	}
+	pushCertificateRespBody, err := utils.FlattenResponse(pushCertificateResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(d.Get("certificate_id").(string))
+	d.Set("targets", flattenResult(pushCertificateRespBody))
+
+	return resourceCcmCertificateRead(ctx, d, meta)
+}
+
+func buildPushCertificateBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"target_service":  d.Get("service"),
+		"target_projects": buildTargetProject(d.Get("targets").(*schema.Set).List()),
+	}
+	return bodyParams
+}
+
+func buildUpdatePushCertificateBodyParams(service string, projectlist []interface{}) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"target_service":  service,
+		"target_projects": buildTargetProject(projectlist),
+	}
+	return bodyParams
+}
+func buildTargetProject(projectlist []interface{}) []string {
+	var projects []string
+	for _, v := range projectlist {
+		projects = append(projects, utils.PathSearch("project_name", v, nil).(string))
+	}
+	return projects
+}
+
+func flattenResult(resp interface{}) []interface{} {
+	curJson := utils.PathSearch("results", resp, make([]interface{}, 0))
+	curArray := curJson.([]interface{})
+	rst := make([]interface{}, 0, len(curArray))
+	for _, v := range curArray {
+		rst = append(rst, map[string]interface{}{
+			"project_name": utils.PathSearch("project_name", v, nil),
+			"cert_id":      utils.PathSearch("cert_id", v, nil),
+		})
+	}
+	return rst
+}
+
+func flattenUpdateResult(d *schema.ResourceData, resp interface{}) []interface{} {
+	targets := d.Get("targets").(*schema.Set).List()
+	rst := make([]interface{}, 0, len(targets))
+	for _, v := range targets {
+		certId := utils.PathSearch("cert_id", v, nil).(string)
+		projectName := utils.PathSearch("project_name", v, nil).(string)
+		if certId == "" {
+			certId = getCertIdFromResult(projectName, resp)
+		}
+		rst = append(rst, map[string]interface{}{
+			"project_name": projectName,
+			"cert_id":      certId,
+			"cert_name":    utils.PathSearch("cert_name", v, nil),
+		})
+	}
+
+	return rst
+}
+
+func getCertIdFromResult(projectName string, resp interface{}) string {
+	var certId = ""
+	curJson := utils.PathSearch("results", resp, make([]interface{}, 0))
+	curArray := curJson.([]interface{})
+	for _, v := range curArray {
+		resName := utils.PathSearch("project_name", v, nil).(string)
```

Comment:
If `project_name` does not exit, it will panic to convert nil to string.

```suggestion
		resName := utils.PathSearch("project_name", v, "").(string)
```